pandas/plotting/_matplotlib/tools.py

```diff
@@ -288,6 +288,57 @@ def _remove_labels_from_axis(axis):
     axis.get_label().set_visible(False)
 
 
+def _has_externally_shared_axis(ax1, compare_axis) -> bool:
```

Comment:
could you pls also annotate `ax1` and `compare_axis`?

Reply:
Done. Will push once I've got the test sorted.